voluncheer/opportunityboard/forms/postanopportunity.py

```diff
@@ -0,0 +1,54 @@
+from django import forms
+from django.contrib.auth.decorators import login_required
+from django.utils import timezone
+
+from opportunityboard.models import Opportunity
+from profiles.models import Organization
+
+
+class PostAnOpportunityForm(forms.ModelForm):
+    """This is the form used for creating a new opportunity for organization users."""
+
+    date = forms.DateTimeField(
+        input_formats=["%d/%m/%Y %H:%M"],
+        widget=forms.DateTimeInput(
+            attrs={
+                "type": "datetime-local",
+            }
+        ),
+    )
+    duration = forms.DurationField(
+        widget=forms.TimeInput(
+            attrs={
+                "type": "time",
+            }
+        ),
+    )
+
+    class Meta:
+        model = Opportunity
+        fields = (
```

Comment:
These are the required fields in the form, yes? If that's true, are we saying that the `date` and `duration` are not required?

Maybe something for the future... We should definitely make those fields required when an organization is publishing an opportunity to the public.

Reply:
The fields attribute in a Model Form is just which fields from the model appear in the form. Date and duration required some customization, hence the declarations outside of `fields`. That said, I added them back into `fields ` as well, because that affects the order of how the fields appear and I prefer them higher up in the form.